mypy/types.py

```diff
@@ -1741,7 +1747,9 @@ def set_typ_args(tp: Type, new_args: List[Type], line: int = -1, column: int = -
     if isinstance(tp, Instance):
         return Instance(tp.type, new_args, line, column)
     if isinstance(tp, TupleType):
-        return tp.copy_modified(items=new_args)
+        fallback_args = [UnionType.make_simplified_union(new_args)]
```

Comment:
If you didn't tell me, I would've never noticed that. Is there any way to protect our codebase from these types of mistakes (where a construct is used too early in the analysis process)?